lib/contentProfileListener.js

```diff
@@ -98,16 +98,36 @@ class ContentProfileImporter extends SHRContentProfileParserListener {
 
     if (currentElement) {
       for (const name of names) {
+        let parentElements = this.getRecursiveBasedOns(currentElement.identifier);
+        
         if (name === 'Value' || name === '_Value') {
           path.push(new Identifier('', '_Value'));
 
-          if (currentElement.value && !(currentElement.value instanceof ChoiceValue)) {
+          // find value from self or most recent ancestor
+          let value = null;
+          for (const id of parentElements) {
+            const el = this._specs.dataElements.findByIdentifier(id);
+            value = el ? el.value : null;
+
+            if (value) {
+              break;
+            }
+          }
+
+          if (value && !(value instanceof ChoiceValue)) {
             currentElement = this._specs.dataElements.findByIdentifier(currentElement.value.effectiveIdentifier);
           } else {
             break; // Exit loop to reach error condition below
           }
         } else {
-          let value = [currentElement.value, ...currentElement.fields].find(field => {
+          // Collect fields from current elements and all parents
+          let fields = new Set();
+          for (const id of parentElements) {
+            const el = this._specs.dataElements.findByIdentifier(id);
+            [el.value, ...el.fields].forEach(field => { if (field) fields.add(field); });
+          }
+          
+          let value = Array.from(fields).find(field => {
             let identifier;
 
             if (!field) {
```

Comment:
I think you don't need to check this anymore because in your new code above you filter out the nulls.